web/src/pill_typeahead.ts

```diff
@@ -72,6 +72,52 @@ export function set_up_user(
     });
 }
 
+export function set_up_stream(
+    $input: JQuery,
+    pills: StreamPillWidget,
+    opts: {
+        help_on_empty_strings?: boolean;
+        update_func?: () => void;
+    },
+): void {
+    const bootstrap_typeahead_input: TypeaheadInputElement = {
+        $element: $input,
+        type: "contenteditable",
+    };
+    opts.help_on_empty_strings ||= false;
+    new Typeahead(bootstrap_typeahead_input, {
+        items: 5,
+        dropup: true,
+        helpOnEmptyStrings: true,
+        source(_query: string): StreamPillData[] {
+            return stream_pill.typeahead_source(pills);
+        },
+        highlighter_html(item: StreamPillData, _query: string): string {
+            return typeahead_helper.render_stream(item);
+        },
+        matcher(item: StreamPillData, query: string): boolean {
+            query = query.toLowerCase();
+            query = query.replaceAll("\u00A0", " ");
+            query = query.trim().slice(1);
```

Comment:
I don't think `slice` is required here since we do not have `#` prefix in this case.

Reply:
Or maybe we can just also match the streams if `#` is present, so we should only slice if first character is `#`.

Reply:
Fixed.